main.c

```diff
@@ -195,11 +195,47 @@ void add_log(struct board_tile board[SIZE][SIZE], int turtle_row[SIZE]) {
     }
 }
 
-void clear_row(struct board_tile board[SIZE][SIZE]) {
-    printf("// TODO clear_row [row]\n");
+void clear_row(
+    struct board_tile board[SIZE][SIZE],
+    int last_coordinate[2]
+) {
+    int row;
+    printf("Enter row to clear: ");
+    scanf("%d", &row);
+
+    // Check if row is out of bounds
+    if (row < 0 || row >= SIZE) {
+        printf("Error: Row %d is out of bounds.\n", row);
+        return;
+    }
+
+    // Check if row is prohibited (LILLYPAD or BANK row)
+    if (row == LILLYPAD_ROW || row == BANK_ROW) {
+        printf("Error: Cannot clear the Lilypad or Bank row.\n");
+        return;
+    }
+
+    // Check if Frogger is on the row
+    for (int col = 0; col < SIZE; col++) {
+        if (board[row][col].occupied == TRUE) {
+            printf("Error: Cannot clear row %d because Frogger is present.\n", row);
+            return;
+        }
+    }
```

Comment:
Using this approach is okay, but you will need to do a loop each time. If the board becomes larger, then you will need to check more columns every time.

But the `last_coordinate` will let us know, where the Frogger is currently since it stores the X at index 0 (i.e. the row) and Y at index 1 (i.e. the column).

So, you just need to check whether the X value matches the input row.

```suggestion
    if (last_coordinate[X] == row) {
        printf("Error: Cannot clear row %d because Frogger is present.\n", row);
        return;
    }
```